app/controllers/api/v1/shelters_controller.rb

```diff
@@ -5,43 +5,56 @@ class Api::V1::SheltersController < ApplicationController
   end
 
   def index
-    @filters = {}
-    @shelters = Shelter.all
+    @shelters, @filters = apply_params(Shelter.all, params)
+  end
+
+  def outdated
+    @outdated, @filters = apply_params(
+      Shelter.where("updated_at < ?", 4.hours.ago).order('updated_at DESC'),
+      params
+    )
+  end
+
+  private
+
+  def apply_params(shelters, params)
```

Comment:
You don't need to pass params here, the params object is accessible anywhere in the controller.

Reply:
Sorry, all my work with functional languages coming through. Will fix.